client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -200,8 +208,17 @@ private Checkpoint completeCheckpoint(String checkpointName) {
 
     @Override
     public void resetReaderGroup(ReaderGroupConfig config) {
+        synchronizer.updateState((state, updates) -> {
+            updates.add(new UpdatingConfig(true));
+        });
+        getThrowingException(controller.updateReaderGroup(scope, groupName, config));
+        ReaderGroupConfig controllerConfig = getThrowingException(controller.getReaderGroupConfig(scope, groupName));
         Map<SegmentWithRange, Long> segments = getSegmentsForStreams(controller, config);
-        synchronizer.updateStateUnconditionally(new ReaderGroupStateInit(config, segments, getEndSegmentsForStreams(config)));
+        synchronizer.updateState((state, updates) -> {
+            if (state.getConfig().getGeneration() < controllerConfig.getGeneration()) {
+                new ReaderGroupStateInit(config, segments, getEndSegmentsForStreams(config), false);
```

Comment:
shouldn't it be using `controllerConfig` here?
Also, if so, then a unit test is missing.

Reply:
This logic has been changed to make sure to account for the case you mentioned [below](https://github.com/pravega/pravega/pull/5439#discussion_r546115261)